### PR TITLE
docs(appengine): add artifact reference for appengine usage

### PR DESCRIPTION
**Pull request by ghost**
This PR summarizes the 3 ways that artifacts can be used in AppEngine deploy stages.

### CODE REVIEW

**Review by lwander — APPROVED** [review]
LGTM, pending @dorbin

File: reference/artifacts/in-appengine/index.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+{%
+  include
+  figure
+  image_path="./gcs_bucket_source_code_artifact.png"
+%}
+
+## Artifacts Referencing Docker Images From GCR
+
+To use an artifact to reference docker images stored in Google Container Registry, set the Server
+Group's Source Type to "Container Image" and set the "Resolve URL" field to "via pipeline artifact":
```
Comment (dorbin) [inline reference/artifacts/in-appengine/index.md]: I think you don't want the colon at the end of this sentence?

**Review by dorbin — COMMENTED** [review]
Scott, can you add an entry to _navigation.yml too?

File: reference/artifacts/in-appengine/index.md (hunk 1 above)
Comment (ghost) [reply reference/artifacts/in-appengine/index.md]: Well spotted; fixed, thanks!

**Comment by ghost** [comment]
I've updated _navigation.yml to include a link to this new page.

**Review by dorbin — APPROVED** [review]
Thanks for the doc, Scott

**Pull request merged**